### PR TITLE
fix(playground): rebuild SelectionPopover examples around a real selection anchor (20ca7572)

### PR DESCRIPTION
**Pull request by stevekinney**
## Summary

The SelectionPopover playground page rendered multiple teal comment buttons scattered across the viewport — every example except `toggled` was open-by-default at hardcoded viewport coords, and viewport-clamping rendered three open at once.

The component is correct (viewport-relative `{x,y}`, native `popover="manual"` top-layer). Fixed the **examples + docs**:

- **basic + keyboard**: the popover now appears only after a real text selection, with `position` computed from the Selection `Range`'s `getBoundingClientRect()` (viewport-relative). The listener is registered at the **document level** via `onMount` with teardown (see review note below), and closes the popover when the selection is collapsed, missing, or outside the surface.
- **null-position + toggled**: start closed, open via an explicit Cinder Button.
- **viewport-clamping**: the three out-of-bounds popovers are gated behind a toggle (`open={showClamping}`, starts false) — nothing scatters on load.
- **README**: real purpose/usage + explicit note that `position` is viewport-relative selection geometry; the usage example uses the correct document-level wiring.
- Added a regression test that statically asserts no example carries a default-open `SelectionPopover` and that selection-driven examples use document-level `selectionchange` (not element-level).

Public API unchanged.

## Review committee

Codex (gpt-5.4, xhigh) + author. Codex caught a **critical bug**: `selectionchange` is a `document` event, so the original `onselectionchange={...}` on the `<article>` element **never fired** — the "select to open" examples were silently broken. Fixed to a document-level listener with `onMount` cleanup (matching the shipped `review-editor` pattern), with full open/close edge-case handling. The regression test was strengthened to handle both tag forms and to guard against the element-level misuse recurring.

## Test plan
- [x] `bun run --filter=cinder test -- selection-popover` — 6 pass
- [x] `bun run --filter=@cinder/playground test` — green
- [x] `components:check` / `typecheck` / `lint` — clean
- Note: pixel-level popover positioning needs a real browser; the open/close flow mirrors shipped review-editor code.

---

> [!NOTE]
> **Low Risk**
> Documentation, playground examples, and static regression tests only; no component API or runtime behavior changes in the library.
> 
> **Overview**
> Fixes the **SelectionPopover** playground and docs so the page no longer shows multiple comment popovers on load, and selection-driven demos actually work.
> 
> **Playground / examples:** Replaces always-open, hardcoded `position` demos with state that starts closed. **Basic** and **keyboard** open only after a real in-surface selection, with `position` from `Range.getBoundingClientRect()` and a document-level `selectionchange` listener (with guards so clicks in the expanded popover do not dismiss it). **Null-position**, **toggled**, and **viewport-clamping** use explicit **Button** triggers; clamping’s three out-of-bounds popovers are gated behind a toggle instead of rendering open immediately.
> 
> **Docs:** Expands the component **README** (purpose, viewport-relative `position`, full usage sample) and aligns `selection-popover.examples.json` with the new patterns.
> 
> **Tests:** Adds source-level regression tests that forbid hardcoded `open` on example call sites and require document-level `selectionchange` for selection-driven examples.
> 
> No public **SelectionPopover** API changes.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5f5e8e1ac495f3acc1d50dbef0e898d2c87a8eca. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/playground/src/examples/selection-popover/null-position.example.svelte [hunk 1]
```diff
@@ -1,49 +1,66 @@
...
-    while leaving <code>open</code> true — the popover should disappear because the component calls
-    <code>hidePopover()</code>. <em>Restore position</em> brings it back at the original coordinates.
+  <p style="margin: 0; font-size: var(--cinder-text-sm); color: var(--cinder-text-muted);">
+    Status: open={isOpen}, position={position ? `{x:${position.x},y:${position.y}}` : 'null'}
   </p>
-
-  <SelectionPopover id="null-position-selection-popover" open {position} />
 </div>
+
+<SelectionPopover id={popoverId} open={isOpen} {position} onclose={handleClose} />
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/playground/src/examples/selection-popover/null-position.example.svelte]: Real bug, fixed. The document-level pointerdown→onclose fired before the control button's click handler, closing the popover so the controls (Restore position, Hide) were non-functional. null-position is now purely programmatically controlled (removed the onclose reset); toggled is open-only + disabled-while-open so its only close path is a genuine outside click. basic/keyboard/viewport-clamping were unaffected. Flagged separately: the component has no closeOnOutsideClick/triggerElement opt-out — a real consumer combining external controls with onclose would hit this same race; that's a worthwhile component follow-up but out of this examples task's scope.

File: packages/playground/src/examples/selection-popover/basic.example.svelte [hunk 2]
```diff
@@ -1,46 +1,117 @@
...
+      if (!surfaceElement) return;
+      const anchor = getSelectionAnchor(surfaceElement);
+      if (anchor) {
+        position = anchor;
+        isOpen = true;
+      } else {
+        isOpen = false;
+        position = null;
+      }
+    }
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/playground/src/examples/selection-popover/basic.example.svelte:79]: Real bug, fixed. Mirrored review-editor-impl's two-guard pattern: the selectionchange handler now skips closing if focus is inside the popover (getElementById(popoverId).contains(activeElement)) or if it's expanded (data-cinder-expanded attribute — for Safari, which doesn't move focus to clicked buttons). Both use the already-exposed id + data attribute, no new API. Genuine outside interaction still closes it.
